pkg/identity/cache/local.go

```diff
@@ -191,3 +191,15 @@ func (l *localIdentityCache) GetIdentities() map[identity.NumericIdentity]*ident
 
 	return cache
 }
+
+// close closes the events channel. Since the local identity cache is the
+// writing party, it is responsible for closing the channel.
+func (l *localIdentityCache) close() {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+
+	if l.events != nil {
+		close(l.events)
+		l.events = nil
```

Comment:
If we set m.events to nil (which is different variable compared to removed `m.events` above), we will have to guard all uses of `l.events` in `localIdentityCache` with an `if m.events != nil` statement, otherwise the send statements that previously paniced after close will now block forever.

Reply:
This is already the case (afaict, the only two writes are `lookupOrCreate` and `release`). `events` is nil in most of the tests.

Reply:
Oh, apologies, I missed that!